### PR TITLE
Add axios interceptor

### PR DESCRIPTION
**Pull request by brunomichetti**
- Add an axios interceptor to refresh session of the user

### CODE REVIEW

File: src/helpers/client.js [hunk 1]
```diff
@@ -1,10 +1,49 @@
...
+      error.config.url != 'rest-auth/logout/'
+    ) {
+      if (error.response.status == 401) {
+        const refreshToken = localStorage.getItem('refreshToken');
+        if (refreshToken) {
+          try {
+            const {
+              data: { access: accessToken },
+            } = await axios.post(
+              process.env.REACT_APP_BASE_URL + 'rest-auth/token/refresh/',
```
Comment (tamarasuarezrod) [inline src/helpers/client.js]:
```suggestion
              `${process.env.REACT_APP_BASE_URL} rest-auth/token/refresh/`,
```

File: src/helpers/client.js [hunk 2]
```diff
@@ -1,10 +1,49 @@
...
+  (res) => res,
+  async (error) => {
+    if (
+      error.config.url != 'rest-auth/login/' &&
+      error.config.url != 'rest-auth/registration/' &&
+      error.config.url != 'rest-auth/logout/'
+    ) {
+      if (error.response.status == 401) {
+        const refreshToken = localStorage.getItem('refreshToken');
+        if (refreshToken) {
```
Comment (tamarasuarezrod) [inline src/helpers/client.js:25]: Just a question, are you refreshing the token even if you have a non-expired one? I think you should have an endpoint to validate the token first, don't you?

Comment (brunomichetti) [reply src/helpers/client.js:25]: I refresh only when the response status is 401, that is, unauthorized. So if I am authorized there is no problem about that

**Review by tamarasuarezrod — APPROVED** [review]

File: src/services/target.services.js [hunk 3]
```diff
@@ -1,11 +1,11 @@
 import client from '../helpers/client';
 
 const create = (radius, title, topic, lat, lon) => {
-  const user = JSON.parse(localStorage.getItem('user'));
+  const accessToken = localStorage.getItem('accessToken');
   return client.post(
     'targets/',
     JSON.stringify({ radius, title, topic, lat, lon }),
-    { headers: { Authorization: `Bearer ${user.access_token}` } }
+    { headers: { Authorization: `Bearer ${accessToken}` } }
```
Comment (peter7z) [inline src/services/target.services.js:8]: I think you should remove this and add an interceptor for the requests that automatically adds the token.

By the way, I love how you are managing all of this manually, it'll really help you understand how this stuff works.
Next step might be to sync the token with redux.

**Review by manuelcheveste — APPROVED** [review]

**Pull request merged**